module_analysis/models/ir_module_module.py

```diff
@@ -0,0 +1,186 @@
+        return {
+            'author_ids': [6, 0, []],
+            'module_type_id': False,
+            'python_code_qty': False,
+            'xml_code_qty': 0,
+            'js_code_qty': 0,
+            'css_code_qty': 0,
+        }
+
+    @api.model
+    def _get_module_encoding(self, file_ext):
+        return 'utf-8'
+
+    # Overload Section
+    @api.model
+    def update_list(self):
+        res = super().update_list()
+        if self.env.context.get('analyse_installed_modules', False):
+            self.search([('state', '=', 'installed')]).button_analyse_code()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get('state', False) == 'installed':
+            self.button_analyse_code()
+        elif vals.get('state', False) == 'uninstalled'\
+                and 'module_analysis' not in [x.name for x in self]:
+            self.write(self._get_clean_analyse_values())
+        return res
+
+    # Public Section
+    @api.multi
+    def button_analyse_code(self):
+        IrModuleAuthor = self.env['ir.module.author']
+        IrModuleTypeRule = self.env['ir.module.type.rule']
+        rules = IrModuleTypeRule.search([])
+
+        cfg = self.env['ir.config_parameter']
+        val = cfg.get_param('module_analysis.exclude_directories', '')
+        exclude_directories = [x.strip() for x in val.split(',') if x.strip()]
+        val = cfg.get_param('module_analysis.exclude_files', '')
+        exclude_files = [x.strip() for x in val.split(',') if x.strip()]
+
+        for module in self:
+            _logger.info("Analysing Code for module %s ..." % (module.name))
+
+            # Update Authors, based on manifest key
+            authors = []
+            if module.author and module.author[0] == '[':
+                author_txt_list = safe_eval(module.author)
+            else:
+                author_txt_list = module.author.split(',')
+
+            author_txt_list = [x.strip() for x in author_txt_list]
+            author_txt_list = [x for x in author_txt_list if x]
+            for author_txt in author_txt_list:
+                authors.append(IrModuleAuthor._get_or_create(author_txt))
+
+            author_ids = [x.id for x in authors]
+            module.author_ids = author_ids
+
+            # Update Module Type, based on rules
+            module_type_id = rules._get_module_type_id_from_module(module)
+            module.module_type_id = module_type_id
+
+            # Get Path of module folder and parse the code
+            module_path = get_module_path(module.name)
+
+            # Get Files
+            analysed_datas = self._get_analyse_data_dict()
+            file_extensions = analysed_datas.keys()
+            file_list = self._get_files_to_analyse(
+                module_path, file_extensions, exclude_directories,
+                exclude_files)
+
+            for file_path, file_ext in file_list:
+                file_res = pygount.source_analysis(
```

Comment:
This method call is deprecation, it was fixed in 12.0 branch, can you cherry-pick these commits? 

https://github.com/OCA/server-tools/pull/2036/commits/755569b57ba595f555db289a2fbc9b1e3cb358cf
https://github.com/OCA/server-tools/pull/2038/commits/315f7e8f22ee6f1ef88b423bfc451b900471df34

Reply:
Done Thank you for your feedback :)